packages/eds-demo/src/app/_components/Typography.tsx

```diff
@@ -0,0 +1,235 @@
+
+type FontSize =
+  | "xs"
+  | "sm"
+  | "md"
+  | "lg"
+  | "xl"
+  | "2xl"
+  | "3xl"
+  | "4xl"
+  | "5xl"
+  | "6xl";
+type LineHeight = "normal" | "squished";
+type As =
+  | "p"
+  | "span"
+  | "div"
+  | "h1"
+  | "h2"
+  | "h3"
+  | "h4"
+  | "h5"
+  | "h6"
+  | "label"
+  | "strong"
+  | "em";
+
+interface TypographyProps
+  extends Omit<HTMLAttributes<HTMLElement>, "className"> {
+  children: ReactNode;
+  size?: FontSize;
+  lineHeight?: LineHeight;
+  baselineAligned?: boolean;
+  debug?: boolean;
+  as?: As;
+  className?: string;
+  weight?: "lighter" | "normal" | "bolder";
+}
+
+/**
+ * Typography component using the EDS 2.0 typography system
+ *
+ * Implements a data-attribute driven sizing system with:
+ * - Baseline grid text alignment
+ * - Space toggle based CSS custom properties
+ * - Ratio-based padding calculations for optical alignment
+ *
+ * @example
+ * // Basic usage
+ * <Typography size="md">Body text</Typography>
+ *
+ * @example
+ * // With all props
+ * <Typography
+ *   size="lg"
+ *   lineHeight="squished"
+ *   baselineAligned={false}
+ *   weight="semibold"
+ *   as="h2"
+ * >
+ *   Heading
+ * </Typography>
+ */
+export function Typography({
+  children,
+  size = "md",
+  lineHeight = "normal",
+  baselineAligned = true,
+  debug = false,
+  as: Component = "p",
+  className = "",
+  weight,
+  style,
+  ...props
+}: TypographyProps) {
+  // Build data attributes based on props
+  const dataAttributes = {
+    "data-font-size": size,
+    ...(lineHeight === "squished" && { "data-line-height": "squished" }),
+    ...(!baselineAligned && { "data-not-baseline-aligned": "" }),
+    ...(debug && { "data-debug": "" }),
+  };
+
+  // Add font-weight if specified
+  const fontWeights = {
+    lighter: 300,
+    normal: 400,
+    bolder: 500,
+  };
+
+  const computedStyle: CSSProperties = {
+    ...style,
+    ...(weight && { fontWeight: fontWeights[weight] }),
+  };
+
+  return (
+    <Component
+      className={`type ${className}`}
+      style={computedStyle}
+      {...dataAttributes}
+      {...props}
+    >
+      {children}
+    </Component>
+  );
+}
+
+// ============================================
+// Preset components for common use cases
+// ============================================
+
+interface HeadingProps extends Omit<TypographyProps, "as"> {
+  level?: 1 | 2 | 3 | 4 | 5 | 6;
+}
+
+/**
+ * Heading component for headings (H1-H6)
+ *
+ * Defaults to semibold weight and squished line-height for compact headings
+ *
+ * @example
+ * <Heading level={1}>Main heading</Heading>
+ * <Heading level={2} weight="bolder">Subheading</Heading>
+ */
+export function Heading({
+  children,
+  level = 1,
+  size = "lg",
+  lineHeight = "squished",
+  weight = "lighter",
+  ...props
+}: HeadingProps) {
+  return (
+    <Typography
+      as={`h${level}` as As}
+      size={size}
+      lineHeight={lineHeight}
+      weight={weight}
+      {...props}
+    >
+      {children}
+    </Typography>
+  );
+}
```

Comment:
I think we can remove the children from the prop destructure and pass it through rest props. This pattern works well for wrapper components that just forward all props (including children) to a child component.

```suggestion
export function Heading({
  level = 1,
  size = "lg",
  lineHeight = "squished",
  weight = "lighter",
  ...props
}: HeadingProps) {
  return (
    <Typography
      as={`h${level}` as As}
      size={size}
      lineHeight={lineHeight}
      weight={weight}
      {...props}
    />
  );
}
```